openflow13/action.go

```diff
@@ -101,9 +101,80 @@ func DecodeAction(data []byte) Action {
 		a = new(ActionPush)
 	case ActionType_PopPbb:
 		a = new(ActionHeader)
+	case ActionType_Experimenter:
+		// For Experimenter message, the length of action should be at least 10 bytes,
+		// including type(2 byte), length(2 byte), vendor(4 byte), and subtype(2 byte)
+		if len(data) < NxActionHeaderLength {
+			return nil, errors.New("The byte array has wrong size for openflow experimenter message")
+		}
+		v := binary.BigEndian.Uint32(data[4:8])
+		if v == NxExperimenterID {
+			subtype := binary.BigEndian.Uint16(data[8:])
+			switch subtype {
+			case NXAST_RESUBMIT:
+				a = new(NXActionResubmit)
+			case NXAST_SET_TUNNEL:
+			case NXAST_DROP_SPOOFED_ARP:
+			case NXAST_SET_QUEUE:
+			case NXAST_POP_QUEUE:
+			case NXAST_REG_MOVE:
+				a = new(NXActionRegMove)
+			case NXAST_REG_LOAD:
+				a = new(NXActionRegLoad)
+			case NXAST_NOTE:
+			case NXAST_SET_TUNNEL_V6:
+			case NXAST_MULTIPATH:
+			case NXAST_AUTOPATH:
+			case NXAST_BUNDLE:
+			case NXAST_BUNDLE_LOAD:
+			case NXAST_RESUBMIT_TABLE:
+				a = new(NXActionResubmitTable)
+			case NXAST_OUTPUT_REG:
+				a = newNXActionOutputReg()
+			case NXAST_LEARN:
+			case NXAST_EXIT:
+			case NXAST_DEC_TTL:
+				a = new(NXActionDecTTL)
+			case NXAST_FIN_TIMEOUT:
+			case NXAST_CONTROLLER:
+			case NXAST_DEC_TTL_CNT_IDS:
+				a = new(NXActionDecTTL)
+			case NXAST_PUSH_MPLS:
+			case NXAST_POP_MPLS:
+			case NXAST_SET_MPLS_TTL:
+			case NXAST_DEC_MPLS_TTL:
+			case NXAST_STACK_PUSH:
+			case NXAST_STACK_POP:
+			case NXAST_SAMPLE:
+			case NXAST_SET_MPLS_LABEL:
+			case NXAST_SET_MPLS_TC:
+			case NXAST_OUTPUT_REG2:
+				a = new(NXActionOutputReg)
+			case NXAST_REG_LOAD2:
+			case NXAST_CNJUNCTION:
```

Comment:
I am not sure why the "O" is left out of CONJUNCTION.

Reply:
Done.